federation/pkg/federated-controller/cluster/cluster_cache.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cluster
+
+import (
+	federation_v1alpha1 "k8s.io/kubernetes/federation/apis/federation/v1alpha1"
+	kubeCache "k8s.io/kubernetes/pkg/client/cache"
+)
+
+// StoreToClusterLister makes a Store have the List method of the unversioned.ClusterInterface
+// The Store must contain (only) clusters.
+type StoreToClusterLister struct {
```

Comment:
the cache will be used by "scheduler" "controller" , so `cache` locate at `federation/client/cache/` is more better.